runtime/jvmti/jvmtiStartup.c

```diff
@@ -374,7 +373,34 @@ issueAgentOnLoadAttach(J9JavaVM * vm, J9JVMTIAgentLibrary * agentLibrary, const
 		invocationJavaVM = (J9InvocationJavaVM *)vm;
 	}
 
-	rc = agentInitFunction(invocationJavaVM, (char *) options, NULL);
+#if defined(WIN32)
+	{
+		/* agentInitFunction invokes java.instrument/share/native/libinstrument/InvocationAdapter.c
+		 * which expects jarPath in system default code page encoding.
+		 */
+		char *jarPath = NULL;
+		BOOLEAN conversionSucceed = FALSE;
+		int32_t size = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_WINDEFAULTACP, options, strlen(options), NULL, 0);
+		if (size > 0) {
+			size += 1; /* leave room for null */
+			jarPath = j9mem_allocate_memory(size, OMRMEM_CATEGORY_VM);
+			if (NULL != jarPath) {
+				size = j9str_convert(J9STR_CODE_MUTF8, J9STR_CODE_WINDEFAULTACP, options, strlen(options), jarPath, size);
+				if (size > 0) {
+					conversionSucceed = TRUE;
+				}
+			}
+		}
+		if (conversionSucceed) {
+			rc = agentInitFunction(invocationJavaVM, jarPath, NULL);
+		}
+		if (NULL != jarPath) {
+			j9mem_free_memory(jarPath);
+		}
+	}
+#else /* defined(WIN32) */
+	rc = agentInitFunction(invocationJavaVM, options, NULL);
```

Comment:
Can you rework this so there is one path that calls `agentInitFunction`?  it may require a few more ifdefs after the call to handle freeing `jarPath`

Reply:
Updated the PR accordingly. Please have another look.